src/memkind_memtier.c

```diff
@@ -330,50 +329,72 @@ memtier_policy_dynamic_threshold_update_config(struct memtier_memory *memory)
     memory->thres_check_cnt = memory->thres_init_check_cnt;
 }
 
-static int builder_static_ctl_set(struct memtier_builder *builder,
-                                  const char *name, const void *val)
+static inline struct memtier_memory *memtier_memory_init(size_t tier_size,
+                                                         bool threshold_var)
 {
-    log_err("Invalid name: %s", name);
-    return -1;
-}
-
-static int memtier_builder_create_threshold(struct memtier_builder *builder,
-                                            unsigned id)
-{
-    if (builder->thres_size > id) {
-        // nothing to do
-        return 0;
+    if (tier_size == 0) {
+        log_err("No tier in builder.");
+        return NULL;
     }
 
-    struct memtier_threshold_cfg *thres =
-        jemk_realloc(builder->thres, sizeof(*thres) * (id + 1));
+    struct memtier_memory *memory = jemk_malloc(sizeof(struct memtier_memory));
+    if (!memory) {
+        log_err("malloc() failed.");
+        return NULL;
+    }
 
-    if (!thres) {
-        log_err("realloc() failed.");
-        return -1;
+    memory->cfg = jemk_calloc(tier_size, sizeof(struct memtier_tier_cfg));
+    if (!memory->cfg) {
+        log_err("calloc() failed.");
+        jemk_free(memory);
+        return NULL;
+    }
+    if (threshold_var) {
```

Comment:
Sth like `is_dynamic_threshold` would better describe what's going on here. See that you set it to true always when building memtier memory with a dynamic threshold.